langchain/sql_database.py

```diff
@@ -110,6 +112,67 @@ def from_uri(
         _engine_args = engine_args or {}
         return cls(create_engine(database_uri, **_engine_args), **kwargs)
 
+    @classmethod
+    def from_databricks(
+        cls,
+        catalog: str,
+        schema: str,
+        host: Optional[str] = None,
+        api_token: Optional[str] = None,
+        warehouse_id: Optional[str] = None,
+        cluster_id: Optional[str] = None,
+        **kwargs: Any,
+    ) -> SQLDatabase:
+        try:
+            from databricks import sql  # noqa: F401
+        except ImportError:
+            raise ValueError(
+                "databricks-sql-connector package not found, please install with"
+                " `pip install databricks-sql-connector`"
+            )
+        context = None
+        try:
+            from dbruntime.databricks_repl_context import get_context
+
+            context = get_context()
+        except ImportError:
+            pass
+
+        default_host = context.browserHostName if context else None
+        if host is None:
+            host = utils.get_from_env("host", "DATABRICKS_HOST", default_host)
+
+        default_api_token = context.apiToken if context else None
+        if api_token is None:
+            api_token = utils.get_from_env(
+                "api_token", "DATABRICKS_API_TOKEN", default_api_token
```

Comment:
The fallback logic of DATABRICKS_API_TOKEN is enolving

Reply:
it's still possible for api_token to end up as None at the moment (if not set in env and context is None). I'm guessing that would cause downstream issues? could do 

```python
if api_token is None and context is not None:
  api_token = context.apiToken
api_token = api_token or utils.get_from_env("api_token", "DATABRICKS_API_TOKEN")
```

without a default val `utils.get_from_env` will fail if it doesn't find the env var

Reply:
@dev2049 The logic here is by design. It will look up the api_token in following logic:
* method parameter `api_token`
* ENV variable `DATABRICKS_API_TOKEN`
* If in Databricks notebook, there will be a temp token from `context.apiToken`
* If api_token is still None, throw error since it is required. The method `utils.get_from_env` can throw a proper exception.

Reply:
I tried running on local jupyter notebook without setting ENV variable DATABRICKS_API_TOKEN
<img width="1390" alt="image" src="https://github.com/hwchase17/langchain/assets/1097932/9870cccd-8355-4440-a645-0b4a3951b708">

Reply:
ah i missed that get_from_env still throws error if provided default val is None, sounds good!